lib/sycamore/sycamore/utils/pdf.py

```diff
@@ -245,3 +242,152 @@ def read_stderr(fh, q):
 
         t_out.join()
         t_err.join()
+
+
+class StdoutEof:
+    "sentinel class"
+
+
+class StderrEof:
+    "sentinel class"
+
+
+class PdfToImageFiles:
+    """
+    Writes the files (streamed) into file_dir.  Caller is responsible for calling
+    path.unlink() to cleanup the files.
+
+    Note: model service will call this to get images for processing
+    """
+
+    def __init__(self, *, pdf_path: str | Path, file_dir: str | Path, resolution: int = 200) -> None:
+        self.file_dir = Path(file_dir)
+        self.in_cm = False
```

Comment:
please write this out as in_contextmanager.  It's a rarely used variable and this means I don't have to reason through what it's doing.